libs/datamodel/connectors/sql-datamodel-connector/src/postgres_datamodel_connector.rs

```diff
@@ -43,30 +122,195 @@ impl Connector for PostgresDatamodelConnector {
         &self.constructors
     }
 
-    fn parse_native_type(&self, name: &str, args: Vec<u32>) -> Option<NativeTypeInstance> {
+    fn parse_native_type(&self, name: &str, args: Vec<u32>) -> Result<NativeTypeInstance, ConnectorError> {
         let constructor = self.find_native_type_constructor(name);
         let native_type = match name {
+            SMALL_INT_TYPE_NAME => PostgresType::SmallInt,
+            INTEGER_TYPE_NAME => PostgresType::Integer,
             BIG_INT_TYPE_NAME => PostgresType::BigInt,
+            DECIMAL_TYPE_NAME => {
+                if let (Some(first_arg), Some(second_arg)) = (args.get(0), args.get(1)) {
+                    PostgresType::Decimal(*first_arg as u8, *second_arg as u8)
+                } else {
+                    return Err(ConnectorError::from_kind(ErrorKind::ArgumentCountMisMatchError {
+                        native_type: String::from(DECIMAL_TYPE_NAME),
+                        given_count: args.len(),
+                        required_count: 2,
+                    }));
+                }
```

Comment:
I guess there's a nice way to extract this into a helper function. This would allow you to cut down on repitition by a lot.